File: FileStorageBackend/FileStorage.Data/Models/StorageItem.cs
using System;
using System.Collections.Generic;

namespace FileStorage.Data.Models
{
    public class StorageItem
    {
        public Guid Id { get; set; }
        public DateTime CreatedOn { get; set; }
        public string TrustedName { get; set; }
        public string Extension { get; set; }
        public string DisplayName { get; set; }
        public long Size { get; set; }
        public bool IsFolder { get; set; }
        public bool IsPrimaryFolder { get; set; }
        public bool IsRecycled { get; set; }
        public bool IsPublic { get; set; }
        public string RelativePath { get; set; }
        public virtual StorageItem ParentFolder { get; set; }
        public virtual Guid? ParentFolderId { get; set; }
        public virtual User User { get; set; }
        public virtual Guid UserId { get; set; }
        public virtual ICollection<StorageItem> StorageItems { get; set; }
    }
}

File: FileStorage-SPA/src/app/_models/storageitem.ts
import { User } from './user';

export interface StorageItem {
    id?: string;
    displayName: string;
    extension: string;
    size: number;
    isFolder: boolean;
    isRecycled: boolean;
    isPublic: boolean;
    user: User;
    userId: string;
    createdOn: Date;
}
